### DISCUSSION TITLE
Zephyrus G16 2024 Slash Light can't show sleep animation when close lid

### DISCUSSION THREAD

**Discussion by Mckay-Yang** [seerge/g-helper#3145] [General]
There are once a issue and been repaired in version 0.181. And this issue is back in version 0.188.
Origin issue: Zephyrus G16 2024 Incorrect Slash Lighting Preset on Lid Close #2815

**Comment by seerge** [comment]
@Mckay-Yang hello,

I can imagine this is connected with another request https://github.com/seerge/g-helper/issues/3054

Can you try this build and check if both lid close ON / OFF options work as expected here ?
[GHelper.zip](https://github.com/user-attachments/files/17083465/GHelper.zip)

**Reply by Mckay-Yang** [reply]
The sleep animation is back, but seems the Disable on lid close option is not working only when not plug the charger.

**Reply by seerge** [reply]
@Mckay-Yang can you define "not working" ? does animation stay on when you close lid on battery or what ?

**Reply by Mckay-Yang** [reply]
I tried several times at 06:05 PM, and this is the log file if it can help.
[Uploading log.txt…]()

**Reply by Mckay-Yang** [reply]
When using the battery, regardless of whether the "Disable on lid close" option is checked, there is no animation after closing the lid.（Sorry, my English is not very good.

**Reply by seerge** [reply]
@Mckay-Yang yeah. But did it behave differently before (in old builds?) I think it's just a way device preserves battery power by turning it OFF completely on battery :)

**Reply by Mckay-Yang** [reply]
I tried 0.187 its just work like the build you gave to me. Also turn off the light when on battery regardless of whether the "Disable on lid close" option is checked. And, it works normally when plugged in.

**Reply by seerge** [reply]
Ok, but then it means this "lid-close" animation be forced on battery at all :)

**Reply by seerge** [reply]
@Mckay-Yang ok, after digging deeper here I have found what is actually happening

When lid is closed it's pretty much 3 separate states Slash Lighting could be in (not 2)
1. OFF (that's what "disable on lid-close" checkbox does)
2. ON and showing animation you had when lid was open
3. ON but showing special "sleep" (blinking) animation

And the problem that guy in https://github.com/seerge/g-helper/issues/3054 and you here ask for **an opposite things**.

I'm personally keep to keep showing old animation (option 2) as default when lid is closed. As this is more logical.

But to accommodate your request I can allow  "sleep" animation as well, but if an optional config param is set in `%appdata\GHelper\config.json`

```
"slash_sleep" : 1,
```

This build should behave as explained above
[GHelper.zip](https://github.com/user-attachments/files/17088923/GHelper.zip)
